cvmfs/lru.h

```diff
@@ -700,6 +720,66 @@ class LruCache : SingleCopy {
     return counters_;
   }
 
+  /**
+   * Prepares for in-order iteration of the cache entries to perform a filter operatoin.
+   * To ensure consistency, the LruCache must be locked for the duration of the filter operation.
+   */
+  virtual void FilterBegin() {
+    assert(!filter_entry_);
+    Lock();
+    filter_entry_ = &lru_list_;
+  }
+
+  /**
+   * Get the current key and value for the filter operation
+   * @param key Address to write the key
+   * @param value Address to write the value
+   */
+  virtual void FilterGet(Key *key, Value *value) {
+    CacheEntry entry;
+    assert(filter_entry_);
+    assert(!filter_entry_->IsListHead());
+    *key = static_cast<ConcreteListEntryContent *>(filter_entry_)->content();
+    bool rc = this->DoLookup(*key, &entry);
+    assert(rc);
+    *value = entry.value;
+  }
+
+  /**
+   * Advance to the next entry in the list
+   * @returns false upon reaching the end of the cache list
+   */
+  virtual bool FilterNext() {
+    assert(filter_entry_);
+    filter_entry_ = filter_entry_->next;
+    return !filter_entry_->IsListHead();
+  }
+
+ /**
+  * Delete the current cache list entry
+  */
+  virtual void FilterDelete() {
```

Comment:
I'd think that `FilterDelete()` would/should rather perform like an "FilterDeleteAndNext".  Perhaps I misunderstand.  Let's discuss.